### PR TITLE
#34: fix compilation warnings

### LINKED ISSUE #34: fix build errors
Fix compilation errors after recent CI images update.

### PR DESCRIPTION
**Pull request by cz4rs**
fixes #34 

Most of the warnings are unused parameter, mixed sign comparison or unused variable.
For the unused parameters, I went with `[[maybe_unused]]` to be more explicit. Maybe it makes sense to remove some of them completely instead.

### CODE REVIEW

File: src/bvh/collision_object.cpp [hunk 1]
```diff
@@ -115,7 +108,7 @@ namespace bvh
   void collision_object::set_entity_data_impl( const void *_data, std::size_t _element_size )
   {
     const int rank = static_cast< int >( ::vt::theContext()->getNode() );
-    const auto od_factor = m_impl->overdecomposition;
+    const auto od_factor = static_cast< std::size_t >( m_impl->overdecomposition );
```
Comment (cz4rs) [inline src/bvh/collision_object.cpp:111]: Maybe it's better to just make `collision_object::impl::overdecomposition` unsigned (it's currently an `int`). Does it ever make sense for overdecomposition to be negative?

**Comment by cz4rs** [comment]
@nmm0 This is partially taken from #11, but since the changes here are completely independent I think it makes sense to merge it as a separate PR.

File: src/bvh/collision_object.cpp (hunk 1 above)
Comment (nmm0) [reply src/bvh/collision_object.cpp:111]: No it should never be negative, we can make it size_t

Comment (cz4rs) [reply src/bvh/collision_object.cpp]: Done.

**Pull request merged**